### PR TITLE
feat(consume): enable loading of `ethereum/tests/BlockchainTests`

### PR DESCRIPTION
**Pull request by danceratopz**
## 🗒️ Description

Enables loading `ethereum/tests/BlockchainTests` (except for 5 json files that define invalid fields via "bigint").

1. Makes `post_state` an optional field in blockchain fixtures.
2. Adds a list of fixtures to skip in `genindex` (the fixtures here all currently use `bigint` to define invalid fields in json).
3. Updates `hasher` to use `_info["generatedTestHash"]` as a synonym for `_info["hash"]` (this allows `genindex` to check whether the `index.json` is up-to-date for `ethereum/tests`; otherwise it must be regenerated every time).

This requires: https://github.com/ethereum/tests/pull/1380.

Test loading of fixtures into EEST pydantic models with:
```console
genindex -i ~/path/to/ethereum/tests/BlockchainTests/
```
Expected output:
```
Skipping '/home/dtopz/code/github/ethereum/tests/BlockchainTests/GeneralStateTests/stTransactionTest/ValueOverflowParis.json'
Skipping '/home/dtopz/code/github/ethereum/tests/BlockchainTests/InvalidBlocks/bc4895-withdrawals/withdrawalsAmountBounds.json'
Skipping '/home/dtopz/code/github/ethereum/tests/BlockchainTests/InvalidBlocks/bc4895-withdrawals/withdrawalsIndexBounds.json'
Skipping '/home/dtopz/code/github/ethereum/tests/BlockchainTests/InvalidBlocks/bc4895-withdrawals/withdrawalsValidatorIndexBounds.json'
Skipping '/home/dtopz/code/github/ethereum/tests/BlockchainTests/InvalidBlocks/bc4895-withdrawals/withdrawalsAddressBounds.json'
Indexing complete 🦄       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:48
```

The resulting `index.json` file contains 22953 tests.

## 🔗 Related Issues
#572 

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: src/cli/gen_index.py [hunk 1]
```diff
@@ -25,6 +25,17 @@
 
 from .hasher import HashableItem
 
+fixtures_to_skip = set(
```
Comment (spencer-tb) [inline src/cli/gen_index.py:28]:
```suggestion
# TODO: remove when these tests are ported or fixed within ethereum/tests.
fixtures_to_skip = set(
```

**Review by spencer-tb — APPROVED** [review]
LGTM! Amazing that we can run these now :D

**Comment by spencer-tb** [comment]
My only confusion surrounding this is why `hive/consensus` has >100k tests where as we only have ~20k. 

Is it 22953 test files maybe? Will have another look

**Comment by danceratopz** [comment]
> My only confusion surrounding this is why `hive/consensus` has >100k tests where as we only have ~20k.
> 
> Is it 22953 test files maybe? Will have another look

Good question! @dimitry moved non-Cancun tests to "legacy tests" in the ethereum/tests branch being used here. We should try loading those tests too.

**Review by marioevz — APPROVED** [review]
LGTM, thanks!

**Comment by winsvega** [comment]
> > My only confusion surrounding this is why `hive/consensus` has >100k tests where as we only have ~20k.
> > Is it 22953 test files maybe? Will have another look
> 
> Good question! @winsvega moved non-Cancun tests to "legacy tests" in the ethereum/tests branch being used here. We should try loading those tests too.

This is to save time and to be able to edit existing tests without breaking anything.
Passing legacy tests should be enough to be able to sync with cancun mainnet. If not, then the mainnet covers all cases for up to cancun implementation. 

The main tests are on >=Cancun to ensure that from current point on there will be no async hardforks. Thats why we focus on those.

**Comment by spencer-tb** [comment]
Perfect! Yeah this makes a lot of sense. There is no point in maintaining the older fork versions of the tests in my eyes. 

But still a nice to have!

>  We should try loading those tests too.

Agreed!

Happy for you to merge this :)

**Pull request merged**